src/components/views/elements/AppTile.js

```diff
@@ -544,13 +551,18 @@ export default class AppTile extends React.Component {
         if (this.props.userWidget) {
             this._onMinimiseClick();
         } else {
+            let promise;
             if (this.props.show) {
                 // if we were being shown, end the widget as we're about to be minimized.
-                this._endWidgetActions();
+                promise = this._endWidgetActions();
+            } else {
+                promise = Promise.resolve();
             }
-            dis.dispatch({
-                action: 'appsDrawer',
-                show: !this.props.show,
+            promise.then(() => {
```

Comment:
I think we might want the widget to be hidden immediately, as the termination may take up to two seconds, no?

Reply:
Ok, I was worried the iframe might be discarded, but it seems to work.